snap_http/api.py

```diff
@@ -109,6 +109,52 @@ def install_all(names: List[str]) -> SnapdResponse:
     return http.post("/snaps", {"action": "install", "snaps": names})
 
 
+def sideload(
+    file_paths: List[str],
+    *,
+    classic: bool = False,
+    dangerous: bool = False,
+    devmode: bool = False,
+    jailmode: bool = False,
+    system_restart_immediate: bool = False,
+) -> SnapdResponse:
+    """Sideload a snap from the local filesystem.
+
+    :param file_paths: Paths to the snap files to install.
+    :param classic: if true, put snaps in classic mode and disable
+        security confinement
+    :param dangerous: if true, install the given snap files even if there are
+        no pre-acknowledged signatures for them
+    :param devmode: if true, put snaps in development mode and disable
+        security confinement
+    :param jailmode: if true, put snaps in enforced confinement mode
+    :param system_restart_immediate: if true, makes any system restart,
+        immediately and without delay (requires snapd 2.52)
+    """
+    data: Dict[str, Union[str, bool]] = {"action": "install"}
+
+    if classic:
+        data["classic"] = classic
+
+    if dangerous:
+        data["dangerous"] = dangerous
+
+    if devmode:
+        data["devmode"] = devmode
+
+    if jailmode:
+        data["jailmode"] = jailmode
+
+    if system_restart_immediate:
+        data["system-restart-immediate"] = system_restart_immediate
+
+    files = []
+    for file_path in file_paths:
+        files.append(FileUpload(name="snap", path=file_path))
```

Comment:
Is there any concern when we have multiple `FileUpload` instances with the same `name`?

Reply:
I don't think so, the API does it that way:

> Snaps can be sideloaded by passing the snap content in a multipart/form-data request with one or more files named “snap”.

```
POST /v2/snaps HTTP/1.1
Host:
Content-Type: multipart/form-data; boundary=foo
Content-Length: 20638

--foo
Content-Disposition: form-data; name="devmode"

true
--foo
Content-Disposition: form-data; name="snap"; filename="hello-world_27.snap"

<20480 bytes of snap file data>
--foo
Content-Disposition: form-data; name="snap"; filename="other-snap_1.snap"

<20480 bytes of snap file data>
--foo
```

https://snapcraft.io/docs/snapd-api#heading--snaps